src/machine/image-varlink.c

```diff
@@ -86,3 +94,87 @@ int vl_method_update_image(sd_varlink *link, sd_json_variant *parameters, sd_var
 
         return sd_varlink_reply(link, NULL);
 }
+
+int vl_method_clone_image(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        static const sd_json_dispatch_field dispatch_table[] = {
+                { "name",     SD_JSON_VARIANT_STRING,  sd_json_dispatch_const_string, offsetof(ImageUpdateParameters, name),      SD_JSON_MANDATORY },
+                { "newName",  SD_JSON_VARIANT_STRING,  sd_json_dispatch_const_string, offsetof(ImageUpdateParameters, new_name),  SD_JSON_MANDATORY },
+                { "readOnly", SD_JSON_VARIANT_BOOLEAN, sd_json_dispatch_tristate,     offsetof(ImageUpdateParameters, read_only), 0 },
+                VARLINK_DISPATCH_POLKIT_FIELD,
+                {}
+        };
+
+        Manager *manager = ASSERT_PTR(userdata);
+        _cleanup_close_pair_ int errno_pipe_fd[2] = EBADF_PAIR;
+        ImageUpdateParameters p = {};
```

Comment:
`.read_only = -1`, no?

Reply:
Not necessary. See where it's used. Setting it to -1 won't have effect

Reply:
FTR. I should have used `sd_json_dispatch_stdbool` instead `sd_json_dispatch_tristate`. But to re-use `struct ImageUpdateParameters` I kept tristate.

Reply:
i'd still initialize it. just feels less surprising.

Reply:
> i'd still initialize it. just feels less surprising.

and then @yuwata comes with his demand of "not necessary initialization". I faced already it :-)